pkg/kubelet/cm/qos_container_manager_linux.go

```
@@ -0,0 +1,266 @@
+
+	return nil
+}
+
+// retrySetMemoryReserve checks for any QoS cgroups over the limit
+// that was attempted to be set in the first Update() and adjusts
+// their memory limit to the usage to prevent further growth.
+func (m *qosContainerManagerImpl) retrySetMemoryReserve(configs map[v1.PodQOSClass]*CgroupConfig, percentReserve int64) error {
+	// Unreclaimable memory usage may already exceeded the desired limit
+	// Attempt to set the limit near the current usage to put pressure
+	// on the cgroup and prevent further growth.
+	for qos, config := range configs {
+		stats, err := m.cgroupManager.GetResourceStats(config.Name)
+		if err != nil {
+			return err
+		}
+		usage := stats.MemoryStats.Usage
+
+		// Because there is no good way to determine of the original Update()
+		// on the memory resource was successful, we determine failure of the
+		// first attempt by checking if the usage is above the limit we attempt
+		// to set.  If it is, we assume the first attempt to set the limit failed
+		// and try again setting the limit to the usage.  Otherwise we leave
+		// the CgroupConfig as is.
+		if configs[qos].ResourceParameters.Memory != nil && usage > *configs[qos].ResourceParameters.Memory {
+			configs[qos].ResourceParameters.Memory = &usage
+		}
+	}
+
+	return nil
+}
+
+func (m *qosContainerManagerImpl) UpdateReserve() error {
+	if len(m.QOSReserveRequests) == 0 {
+		return nil
+	}
+
+	qosConfigs := map[v1.PodQOSClass]*CgroupConfig{
+		v1.PodQOSBurstable: &CgroupConfig{
+			Name:               CgroupName(m.qosContainersInfo.Burstable),
+			ResourceParameters: &ResourceConfig{},
+		},
+		v1.PodQOSBestEffort: &CgroupConfig{
+			Name:               CgroupName(m.qosContainersInfo.BestEffort),
+			ResourceParameters: &ResourceConfig{},
+		},
+	}
+
+	for resource, percentReserve := range m.QOSReserveRequests {
+		switch resource {
+		case v1.ResourceMemory:
+			m.setMemoryReserve(qosConfigs, percentReserve)
```

you appear to ignore the error that this can throw, why?